server/models/pokemon.ts

```diff
@@ -0,0 +1,157 @@
+import { Model, RelationMappings, Transaction } from 'objection';
+import Item from './item';
+
+export enum PokemonType {
+  normal = 'normal',
+  grass = 'grass',
+  fire = 'fire',
+  water = 'water',
+  electric = 'electric',
+  psychic = 'psychic',
+  ghost = 'ghost',
+  dark = 'dark',
+  fairy = 'fairy',
+  rock = 'rock',
+  ground = 'ground',
+  steel = 'steel',
+  flying = 'flying',
+  fighting = 'fighting',
+  bug = 'bug',
+  ice = 'ice',
+  dragon = 'dragon',
+  poison = 'poison',
+}
+
+export interface IPokemonCreateFields {
```

Comment:
```suggestion
export interface IPokemonCreateInput {
```
for naming consistency, since you've named all the other interfaces `Input`